ground/gcs/src/plugins/boards_dronin/seppuku.cpp

```diff
@@ -0,0 +1,198 @@
+/**
+ ******************************************************************************
+ * @file       seppuku.cpp
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2016
+ * @addtogroup GCSPlugins GCS Plugins
+ * @{
+ * @addtogroup Boards_dRonin dRonin board support plugin
+ * @{
+ * @brief Supports dRonin board configuration
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#include <extensionsystem/pluginmanager.h>
+
+#include "seppuku.h"
+#include "hwseppuku.h"
+
+/**
+ * @brief Seppuku:Seppuku
+ *  This is the Seppuku board definition
+ */
+Seppuku::Seppuku(void)
+{
+    ExtensionSystem::PluginManager *pm = ExtensionSystem::PluginManager::instance();
+    uavoManager = pm->getObject<UAVObjectManager>();
+
+    boardType = 0xA1;
+
+    USBInfo board;
+    board.vendorID = 0x20A0;
+    board.productID = 0x4250;
+
+    setUSBInfo(board);
+
+    channelBanks.resize(4);
+    channelBanks.clear();
+    channelBanks << (QVector<int>() << 1 << 2);      // TIM8
+    channelBanks << (QVector<int>() << 3);           // TIM14
+    channelBanks << (QVector<int>() << 4 << 5 << 6); // TIM3
+    channelBanks << (QVector<int>() << 7 << 8);      // TIM5
+}
+
+Seppuku::~Seppuku()
+{
+}
+
+QString Seppuku::shortName()
+{
+    return QString("Seppuku");
+}
+
+QString Seppuku::boardDescription()
+{
+    return QString("Seppuku running dRonin firmware");
+}
+
+bool Seppuku::queryCapabilities(BoardCapabilities capability)
+{
+    switch (capability) {
+    case BOARD_CAPABILITIES_GYROS:
+    case BOARD_CAPABILITIES_ACCELS:
+    case BOARD_CAPABILITIES_MAGS:
+    case BOARD_CAPABILITIES_BAROS:
+    case BOARD_CAPABILITIES_OSD:
+    case BOARD_CAPABILITIES_UPGRADEABLE:
+        return true;
+    default:
+        break;
+    }
+    return false;
+}
+
+bool Seppuku::isInputConfigurationSupported(InputType type)
+{
+    Q_UNUSED(type)
+    return false;
```

Comment:
Shouldn't this "false" be a "true"?
Else you wouldn't be needing the set- and getInputType functions.